target/linux/ipq40xx/patches-5.15/105-add-linux-spidev-compatible-em3581.patch

```diff
@@ -0,0 +1,19 @@
+Add the "siliconlabs,em3581" compatible in spidev.
+
+This is needed for all devices using a Silicon Labs EM3581 chip 
+connected to the SPI interface to manage a Zigbee radio.
+
+Signed-off-by: Vincent Tremblay <vincent@vtremblay.dev>
+
+Index: linux-5.15.84/drivers/spi/spidev.c
+===================================================================
+--- linux-5.15.84.orig/drivers/spi/spidev.c
++++ linux-5.15.84/drivers/spi/spidev.c
+@@ -697,6 +697,7 @@ static const struct of_device_id spidev_
+ 	{ .compatible = "cisco,spi-petra" },
+ 	{ .compatible = "micron,spi-authenta" },
+ 	{ .compatible = "siliconlabs,si3210" },
++	{ .compatible = "siliconlabs,em3581" },
```

Comment:
Can you send this upstream?

Otherwise we are gonna be stuck forever with maintaining the patch

Reply:
Sure, I will send the patch as soon as possible

Reply:
The patch is [sent](https://patchwork.kernel.org/project/spi-devel-general/patch/20221224041825.171345-1-vincent@vtremblay.dev/).

Reply:
Please follow-up on the bindings request, you can just add it to trivial-devices

Reply:
Yep, so small update; the patch will be applied for 6.3
https://git.kernel.org/pub/scm/linux/kernel/git/broonie/spi.git/commit/?h=for-6.3
https://git.kernel.org/pub/scm/linux/kernel/git/broonie/spi.git/commit/?id=54c467b314b6c2afeb7093d86886103392e3499c

I changed the manufacturer name from `siliconlabs` to `silabs` to match existing bindings in the kernel and I have updated this PR accordingly.

There's also another "spidev" patch for `siliconlabs,si3210` in target/linux/generic/pending-5.15 ([see here](https://github.com/openwrt/openwrt/blob/master/target/linux/generic/pending-5.15/130-add-linux-spidev-compatible-si3210.patch)) that I could not find upstream, so I [submitted it](https://patchwork.kernel.org/project/spi-devel-general/list/?series=707158). If it is approved and applied I will open another PR to rename the manufacturer in the other DTS.

Reply:
That is great, thanks for taking care of it.

Reply:
Please, see this [PR](https://github.com/openwrt/openwrt/pull/11667/)

Reply:
Please, see this [PR](https://github.com/openwrt/openwrt/pull/11678) for the patches. Thanks!

Reply:
The patches were merge on master, I have rebased this branch and removed [target/linux/ipq40xx/patches-5.15/105-add-linux-spidev-compatible-em3581.patch](https://github.com/openwrt/openwrt/pull/11602/files#diff-5508d3c8b7377e30a57a66cab6b213537c560dd4f04789cac937614d457ed8f5)